### PR TITLE
feat: add docker compose

### PR DESCRIPTION
**Pull request by geekyharsh05**
This pr setups:
-  Docker compose
-  Integrates postgres db in the compose
-  Integrates prisma-studio

This pr solves the issue #28 

I have tested it on my end. Verified that services start correctly using `docker-compose up`.

Please review the Docker Compose configuration and let me know if any adjustments are needed.

### CODE REVIEW

**Comment by SujithThirumalaisamy** [comment]
Remove the lock files.

**Comment by geekyharsh05** [comment]
@SujithThirumalaisamy Done!

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,15 @@
-FROM node:22.6.0
```
[the remark this answers is not in the document]
Comment (geekyharsh05) [reply Dockerfile:1]: Ok, I will do that. Thank you!

**Comment by geekyharsh05** [comment]
@SujithThirumalaisamy I have added the Dockerfile for the development mode.

**Comment by geekyharsh05** [comment]
@hkirat Can you please review this?

**Comment by SujithThirumalaisamy** [comment]
you removed the whole lock file. Re-add it and also resolve the conflicts

**Comment by geekyharsh05** [comment]
@SujithThirumalaisamy Sorry, for the trouble i am doing it for the first time!

**Comment by SujithThirumalaisamy** [comment]
Thankyou!

**Pull request merged**